cube/src/main/java/zone/cogni/asquare/cube/convertor/data2shacl/ShaclGenerator.java

```diff
@@ -458,12 +481,17 @@ private List<String> selectForTypeAndProperty(RdfStoreService rdfStoreService,
                                                 String fileName,
                                                 Resource type,
                                                 Resource property) {
+    List<Map<String, RDFNode>> rows = getRowsForTypeAndProperty(rdfStoreService, fileName, type, property);
+    return paginatedQuery.convertSingleColumnUriToStringList(rows);
+  }
+
+  private List<Map<String, RDFNode>> getRowsForTypeAndProperty(RdfStoreService rdfStoreService, String fileName, Resource type, Resource property) {
     Map<String, String> parameters = Map.of("type", type.getURI(),
                                             "property", property.getURI());
     String query = spelService.processTemplate(getResource(fileName),
                                                parameters);
     List<Map<String, RDFNode>> rows = paginatedQuery.select(rdfStoreService, query);
-    return paginatedQuery.convertSingleColumnUriToStringList(rows);
+    return rows;
```

Comment:
direct return of paginatedQuery.select(rdfStoreService, query); ?

Reply:
what else? there is only 1 select in PaginatedQuery... and it returns just that

Reply:
just 1 line:
return paginatedQuery.select(rdfStoreService, query);

Reply:
ah. like that 😄 
you should have proposed a code change 😸